dask/dataframe/groupby.py

```diff
@@ -514,7 +514,11 @@ def _non_agg_chunk(df, *by, key, dropna=None, observed=None, **kwargs):
             # If we found any categoricals, append unobserved values to the end of the
             # frame.
             new_cats = full_index[~full_index.isin(result.index)]
-            empty = pd.DataFrame(pd.NA, index=new_cats, columns=result.columns)
+            empty_data = {
+                col: pd.Series([None], name=col, dtype=result[col].dtype)
+                for col in result.columns
+            }
+            empty = pd.DataFrame(empty_data, index=new_cats)
```

Comment:
I'm wondering if this can still lead to all-`pd.NA` entries? It looks like this works for the current failing test 

```python
FAILED dask/dataframe/tests/test_groupby.py::test_groupby_aggregate_categorical_observed[median-disk-unobserved-groupby1-unordererd-unknown] - FutureWarning: The behavior of DataFrame concatenation with all-NA entries is deprecated. In a future version, this will no longer exclude all-NA columns when determining the result dtypes. To retain the old behavior, cast the all-NA columns to the desired dtype before the concat operation.
```

because `dtype="float64"`, so `None` gets converted to `np.nan`

```python
In [1]: import pandas as pd

In [2]: pd.Series([None], dtype="float64")
Out[2]:
0   NaN
dtype: float64
```

But what about other dtypes like `Float64` where `None` gets converted to `pd.NA`? 

```python
In [3]: pd.Series([None], dtype="Float64")
Out[3]:
0    <NA>
dtype: Float64
```

Reply:
This should be fine. Previously if you had one all NA Series with dtype object and an Integer Series it would ignore the object dtype instead of casting to object. This isn't a problem anymore when the all NA Series has the appropriate dtype, which is what this PR achieves

Reply:
The problem is not all NA entries, but not setting an explicit `dtype` in empty dataframe. This snippet is setting it to original dtype.

Reply:
Ah, grand. Thanks for clarifying 👍